moveit_planners/trajopt/src/trajopt_planner_manager.cpp

```diff
@@ -37,104 +37,101 @@
-                                                            const planning_interface::MotionPlanRequest& req,
-                                                            moveit_msgs::MoveItErrorCodes& error_code) const override
-  {
-    ROS_INFO(" ======================================= getPlanningContext() is called ");
-    error_code.val = moveit_msgs::MoveItErrorCodes::SUCCESS;
+    bool canServiceRequest(const planning_interface::MotionPlanRequest& req) const override
+    {
+      return req.trajectory_constraints.constraints.empty();
+    }
 
-    if (req.group_name.empty())
+    std::string getDescription() const override
     {
-      ROS_ERROR("No group specified to plan for");
-      error_code.val = moveit_msgs::MoveItErrorCodes::INVALID_GROUP_NAME;
-      return planning_interface::PlanningContextPtr();
+      return "TrajOpt";
     }
 
-    if (!planning_scene)
+    void getPlanningAlgorithms(std::vector<std::string>& algs) const override
     {
-      ROS_ERROR("No planning scene supplied as input");
-      error_code.val = moveit_msgs::MoveItErrorCodes::FAILURE;
-      return planning_interface::PlanningContextPtr();
+      algs.clear();
+      algs.push_back("trajopt");
     }
 
-    // create PlanningScene using hybrid collision detector
-    planning_scene::PlanningScenePtr ps = planning_scene->diff();
+    planning_interface::PlanningContextPtr getPlanningContext(const planning_scene::PlanningSceneConstPtr& planning_scene,
+                                                              const planning_interface::MotionPlanRequest& req,
+                                                              moveit_msgs::MoveItErrorCodes& error_code) const override
+    {
+      ROS_INFO(" ======================================= getPlanningContext() is called ");
+      error_code.val = moveit_msgs::MoveItErrorCodes::SUCCESS;
+
+      if (req.group_name.empty())
+      {
+        ROS_ERROR("No group specified to plan for");
+        error_code.val = moveit_msgs::MoveItErrorCodes::INVALID_GROUP_NAME;
+        return planning_interface::PlanningContextPtr();
+      }
 
-    // set FCL for the collision
-    ps->allocateCollisionDetector(collision_detection::CollisionDetectorAllocatorFCL::create());
+      if (!planning_scene)
+      {
+        ROS_ERROR("No planning scene supplied as input");
+        error_code.val = moveit_msgs::MoveItErrorCodes::FAILURE;
+        return planning_interface::PlanningContextPtr();
+      }
 
-    // retrieve and configure existing context
-    const TrajOptPlanningContextPtr& context = planning_contexts_.at(req.group_name);
+      // create PlanningScene using hybrid collision detector
+      planning_scene::PlanningScenePtr ps = planning_scene->diff();
 
-    ROS_INFO(" ======================================= context is made ");
+      // set FCL for the collision
+      ps->getCollisionDetectionPlugin()->initialize("FCL");
```

Comment:
Why this change? I don't know the difference between:

`ps->allocateCollisionDetector(collision_detection::CollisionDetectorAllocatorFCL::create());`

and

`ps->getCollisionDetectionPlugin()->initialize("FCL");`